### ISSUE TITLE
map.kml  file location

### ISSUE THREAD

**Issue by DjAnu** [brchri/tesla-geogdo#11]
Hi, Sorry couldn't figure out where to place .kml file and how to define path. I saved in same folder where config.yml is located and gave part as
    kml_file: volume1/docker/tesla-geogdo/map.kml

but I am getting following error

![image](https://github.com/brchri/tesla-geogdo/assets/5419410/63b246b7-f39e-494c-b43a-31a995763f87)

**Comment by DjAnu** [comment]
Nevermind I used /app/config/map.xml and container loaded successfully.

**Comment by DjAnu** [comment]
Ok seems like I didn't get any error but upon changing the LAT back and forth it didn't fire any command either :-(.

**Comment by brchri** [comment]
This probably means that there's something wrong with the formatting of your kml file. Unfortunately the kml spec isn't very well defined and different tools generate slightly different formats and it's difficult to account for those variations.

Because of that, the easiest way to use the kml file is to use the [example](https://github.com/brchri/tesla-geogdo/blob/main/resources/polygon_map.kml) in the repo and replace the lat/long values there with the ones generated for your own, and then just use that.

Hope that helps.

**Comment by DjAnu** [comment]
Ahaa, so that was the issue. I just saved the .kml file from geojson and was wondering there are no boundary names in my file. I'll use your format, hopefully that will do the trick. Should I set the file location as /app/config/map.xml ?

**Comment by brchri** [comment]
If you have it located in the same folder as your `config.yml` file, then yes that path should work.

**Comment by DjAnu** [comment]
Alright, seems like its all good now. Thanks a lot for your support. You have done such a great job with this App. I am still trying to find an easier way to automate garage door for my wife's GLC. I wish I could use your polygon geofence with other trigger options. Maybe an ibeacon in the car that feeds coordinates to MQTT can be used universally for any other car than Tesla. I know Tesla feeds coordinates pretty often. For now I am thinking of using Owntrack with slightly tighter geofence to auto-open. Auto-close would likely not work due to slower update frequency. If you can think of any way to implement your automation for a universal use that is not dependent on Tesla / Teslamate, Please let me know.....

Again, Thanks so much for your work,

**Comment by brchri** [comment]
Thanks for the kudos, appreciate it. 😸 I mean yeah this could be adapted to other triggers, it's a bit hardcoded to expect teslamate topics right now but it could be modified to work with any mqtt topics if you can feed data that way, and that's probably the most elegant way to do it. Another option could be potentially to open an http listener service on the app and receive webhook updates from some source, but mqtt is a better/cleaner option I think.

**Comment by DjAnu** [comment]
Owntrack ios App does feed the coordinates to MQTT or HTTP. However it's not ideal to trigger Garage open/close just based on phone location. 
Is it possible to trigger an HTTP action like GET or POST command, something like :

https://cloud.hubitat.com/api/c2394adc/apps/195/devices/[DeviceID]/commands?access_token=b04b

This will open up the possibility to literally control anything with any home automation Platform. I am assuming I can still use your http type GDO config to accomplish the command thing. I just need to know how can I utilize the coordinate feeds from Owntrack. Can you help me with that ? I am willing to be a test guineapig and can write some instructions for home-automation forums. I know this will make a large group of people very very happy. Here is link for Owntracks : 

https://owntracks.org/

**Comment by brchri** [comment]
It is possible to trigger (simple) http commands if you use the `http` opener type (see example [here](https://github.com/brchri/tesla-geogdo/blob/main/examples/config.polygon.http.yml#L54-L91)). It wouldn't be able to handle anything sophisticated like auth flows, but if you just need to do a POST or GET request with a predefined API auth token or something, it should work for that.

**Comment by brchri** [comment]
I just reread your message and saw you already discovered the http type, and your actual question was around getting coordinates from onetrack. I have no idea how to retrieve those, does it publish them to a location in the cloud where you can retrieve them or something? Or do you define an endpoint to publish the coordinates to?

**Comment by DjAnu** [comment]
it can actually feed into MQTT or HTTP. I am currently using the driver below that uses HTTP method:
https://community.hubitat.com/t/release-owntracks-presence/53419

But I just saw someone had created a driver that uses MQTT method which seem to have more capabilities. I'll switch to that.

https://community.hubitat.com/t/owntracks-mqtt-driver/22883

**Comment by DjAnu** [comment]
seems like i'll need to configure mosquitto with some security so I can open it to accept connections from outside the home but I think I can handle that part if you can help with triggering part based on locations from Owntrack. Thanks so much.

**Comment by brchri** [comment]
Alright I made a quick and dirty version that should support agnostic mqtt triggers. If it works for you then I'll make it cleaner, but I didn't want to put too much effort into it before proving the concept. To test it, use the image `brchri/tesla-geogdo:enable-agnostic-triggers`, and use something like this for your global config:

```yaml
global:
  mqtt_settings:
    connection:
      host: localhost
      port: 1883
      client_id: tesla-geogdo
      user: mqtt_user
      pass: mqtt_pass
      use_tls: false
      skip_tls_verify: false
    lat_topic: some/topic/latitude
    lng_topic: some/topic/longitude
  cooldown: 5 # minutes to wait after operating garage before allowing another garage operation (set to 0 or omit to disable)
```

Good luck, let me know how it goes.

**Comment by DjAnu** [comment]
Wow you are truly a gem. Any chance you are on Reditt chat ?

**Comment by brchri** [comment]
Nah I'm about to log off for the night.

**Comment by brchri** [comment]
I just published a new image btw, I found a bug in the original, so if you pulled already, you might need to pull again.

**Comment by DjAnu** [comment]
No worries I am creating another container with your new image. How do I
set it use data from Owntracks. Maybe we can chat sometime when you got few
minutes. Truly appreciate your help.

**Comment by DjAnu** [comment]
pulling it now.

**Comment by brchri** [comment]
If you look at the config example I posted earlier, there's a `lat_topic` and `lng_topic` that you would fill out with whatever topics owntracks is publishing to.

**Comment by DjAnu** [comment]
Wait, I figured creating a new container won't work since it's still mapping files to /app/config. I don't want to touch Tesla-geogdo while its working great.

**Comment by DjAnu** [comment]
> If you look at the config example I posted earlier, there's a `lat_topic` and `lng_topic` that you would fill out with whatever topics owntracks is publishing to.

Okay...Let me figure that out first. Thanks.

**Comment by DjAnu** [comment]
So how do i run a new container without touching tesla-geogdo

**Comment by brchri** [comment]
Just create another entry in your docker compose file with a different name and folder mapping and put the new config file in there.

**Comment by DjAnu** [comment]
You mean something like this ? Mount path as appxx/config ?

tesla-geogdo-agnostic:
    image: brchri/tesla-geogdo:enable-agnostic-triggers
    container_name: tesla-geogdo-agnostic
    environment:
      - TZ=America/Los_Angeles
      - DEBUG=false
    volumes:
      - /volume1/docker/tesla-geogdo-agnostic:/appxx/config:ro
    depends_on:
      - mosquitto
      - teslamate
    restart: on-failure:5

**Comment by DjAnu** [comment]
Okay, I see how Owntracks is reporting topics. I need to figure out how to safely expose MQTT for outside network.

**Comment by brchri** [comment]
No leave the path /app/config. It's a separate container so the target paths are completely unique to it. As long as the source path is different you're good.

**Comment by DjAnu** [comment]
Cool pulling the image now

**Comment by brchri** [comment]
> Okay, I see how Owntracks is reporting topics. I need to figure out how to safely expose MQTT for outside network.

You could just use a VPN.

**Comment by DjAnu** [comment]
Getting the following error, tried to remove the last 2 lines with car, then no cars found

![image](https://github.com/brchri/tesla-geogdo/assets/5419410/4db423a8-08c2-4533-ae18-8c42c0a2b664)

**Comment by brchri** [comment]
Show your config?

**Comment by brchri** [comment]
Check your spacing matches up with the example above. It needs the exact same lead spacing shown. It looks like it's not parsing your `lat_topic` and `lng_topic`.

**Comment by DjAnu** [comment]
![image](https://github.com/brchri/tesla-geogdo/assets/5419410/26f6e4bd-e0e9-4d5b-8a9a-649ca5c63ffa)

**Comment by brchri** [comment]
That seems right. I'll take a look tomorrow. 

One note - if you're running this alongside the other geogdo container, you should change those mqtt client IDs to make sure they're not the same as the other container or it will cause issues.

**Comment by DjAnu** [comment]
Alright I am trying to set up another MQTT server on my synology that I can expose to internet or I'll try to find a way to password protect existing server and then set up teslamate and other clients to use user/passwd.

**Comment by brchri** [comment]
I believe I've fixed the problem and pushing another docker image now.

**Comment by DjAnu** [comment]
Alright. Thanks.

**Comment by DjAnu** [comment]
![image](https://github.com/brchri/tesla-geogdo/assets/5419410/44f276d2-d284-43db-9132-8bd04d8a4e0b)

**Comment by brchri** [comment]
This was implemented in v2.0.0, closing issue.
